extensions/positron-supervisor/src/KallichoreAdapterApi.ts

```diff
@@ -705,4 +730,35 @@ export class KCApi implements KallichoreAdapterApi {
 
 		throw new Error(`Kallichore server not found (expected at ${embeddedBinary})`);
 	}
+
+	/**
+	 * Reconnects to the active session, if one exists. Primarily useful as a
+	 * troubleshooting tool.
+	 */
+	async reconnectActiveSession() {
+		// Get the foreground session from the Positron API
+		const session = await positron.runtime.getForegroundSession();
+		if (!session) {
+			vscode.window.showInformationMessage(vscode.l10n.t('No active session to reconnect to'));
+			return;
+		}
+
+		// Find the session in our list
+		const kallichoreSession = this._sessions.find(s => s.metadata.sessionId === session.metadata.sessionId);
+		if (!kallichoreSession) {
+			vscode.window.showInformationMessage(vscode.l10n.t('Active session {0} not managed by the kernel supervisor', session.metadata.sessionName));
+			return;
+		}
+
+		// Ensure the session is still active
+		if (kallichoreSession.runtimeState === positron.RuntimeState.Exited) {
+			vscode.window.showInformationMessage(vscode.l10n.t('Session {0} is not running', session.metadata.sessionName));
+			return;
+		}
+
+		// Disconnect the session; since the session is active, this should
+		// trigger a reconnect.
+		kallichoreSession.log('Disconnecting by user request', vscode.LogLevel.Info);
```

Comment:
For my understanding, where does this log out to? If a user wanted to find this log output, where would they go? I don't think it goes to the Kernel Supervisor Output

Reply:
You are correct! It logs to the Console output logs for the session (e.g. `Python 3.12.4 Console`).